src/main/scala/akka/contrib/datareplication/ORMultiMap.scala

```diff
@@ -7,21 +7,21 @@ import akka.cluster.{ UniqueAddress, Cluster }
 
 object ORMultiMap {
 
+  val _empty: ORMultiMap[Any] = new ORMultiMap(ORMap.empty)
   /**
    * Provides an empty multimap.
    */
-  val empty: ORMultiMap =
-    new ORMultiMap(ORMap.empty)
-  def apply(): ORMultiMap = empty
+  def empty[A]: ORMultiMap[A] = _empty.asInstanceOf[ORMultiMap[A]]
+  def apply(): ORMultiMap[Any] = _empty
 
   /**
    * Java API
    */
-  def create(): ORMultiMap = empty
+  def create[A](): ORMultiMap[A] = empty[A]
 
   def unapply(value: Any): Option[Map[String, Set[Any]]] = value match {
```

Comment:
Good catch. I think the type should be as specific as possible:

```
scala> case class Foo(bar: String)
defined class Foo

scala> :javap -p Foo$
Compiled from "<console>"
public class Foo$ extends scala.runtime.AbstractFunction1<java.lang.String, Foo> implements scala.Serializable {
  public static final Foo$ MODULE$;
  ...
  public scala.Option<java.lang.String> unapply(Foo);
  ...
}
```

I.e. `ORMultiMap` in this case.